### PR TITLE
feat: fetch topology from ipfs

### PR DESCRIPTION
**Pull request by mpetrun5**
Fetch topology from http (IPFS) to avoid dependecy on storage and many variables that it needs.

## Related Issue Or Context

Closes: #31 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: topology/topology.go [hunk 1]
```diff
@@ -82,38 +69,24 @@ type Decrypter interface {
...
-	bucketName   string
-	decrypter    Decrypter
+	url       string
+	decrypter Decrypter
 }
 
 func (t *topologyProvider) NetworkTopology() (NetworkTopology, error) {
-	ctx := context.Background()
-
-	obj, err := t.client.GetObject(ctx, t.bucketName, t.documentName, minio.GetObjectOptions{})
```
Comment (P1sar) [inline topology/topology.go:94]: I thought that previously we were using files/storage API for fetching the topology.
Are we sure that IPFs will always contain that file? I have doubts about it.

File: topology/topology.go [hunk 2]
```diff
@@ -82,38 +69,24 @@ type Decrypter interface {
...
+	resp, err := http.Get(t.url)
```
Comment (P1sar) [inline topology/topology.go:77]: What is that URL supposed to be? Should we keep our IPFS node or fetch from some providers like Cloudflare for example?

**Review by P1sar — CHANGES_REQUESTED** [review]

File: topology/topology.go (hunk 2 above)
Comment (mpetrun5) [reply topology/topology.go:77]: URL is basically an URL to that file on some gateway (ie. ipfs.io)

File: topology/topology.go (hunk 1 above)
Comment (mpetrun5) [reply topology/topology.go:94]: We will still use files to store (get the filecoin integration to pin it), but I would much prefer not fetch from that API as I don't trust it that much and it needs many variables which are annoying to explain and share what they are.

Comment (P1sar) [reply topology/topology.go:94]: but the problem is that I was trying couple of times uploading files to IPFS via the storage or files and then was not be able to download it back with IPFS

**Review by MakMuftic — APPROVED** [review]
@mpetrun5 Did we change these parameters on aws for our dev environment - this is necessary before we merge this. Apart from this, everything looks good.

**Comment by mpetrun5** [comment]
> @mpetrun5 Did we change these parameters on aws for our dev environment - this is necessary before we merge this. Apart from this, everything looks good.

They are updated now.

**Review by P1sar — APPROVED** [review]

**Pull request merged**